src/autoDeleteMessages.ts

```diff
@@ -1,23 +1,53 @@
 import { BotContext } from './context';
 
-export interface AutoDeleteMessage {
-  [messageId: number]: number;
-}
+const expired = (createdAt: number, timeout: number) => {
+  const minute = 60000;
+  return Math.floor((Date.now() - createdAt) / minute) >= timeout;
+};
 
 export const runMessageRecycling = async (
   ctx: BotContext
 ): Promise<void> => {
-  const database = ctx.database;
+  const database = await ctx.loadDatabase();
   const messages = database.getAutoDeleteMessages();
-  const minute = 60000;
-  const recycleTimeout = 1;
-
-  Object.entries(messages).map(async ([id, createdAt]) => {
-    if (
-      Math.floor((Date.now() - createdAt) / minute) >= recycleTimeout
-    ) {
-      await ctx.deleteMessage(+id);
-      await database.removeAutoDeleteMessage(+id);
+  const messageTimeout = 1;
```

Comment:
I don't really understand why the `messageTimeout` here is `1`, but it does works. Could you explain how you designed the `expired` calculation? @AlcidesA

Reply:
Sure, I subtract the current timestamp with the createdAt and divide by a minute to see how many minutes have passed since the message was created, so if the message was created 1+ minute ago, it can be delete.

But I forgot to mention in my PR that 1 minute was set just for tests, I think 5 minutes could work better... What do you think ?

Reply:
I think 5 minutes might be a lot, I was thinking of trying something like 2 minutes initially.

Ideally, this value should come from an environment variable, so I can tweak it on the deployment configuration instead of having to change a hardcoded value for this. I'll push a change that loads this from the configuration.

Reply:
just pushed a change, this now reads it from the `MESSAGE_TIMEOUT_IN_MINUTES` env var after it is parsed during config creation